ugs-core/src/com/willwinder/universalgcodesender/listeners/ControllerListener.java

```diff
@@ -39,7 +40,14 @@ This file is part of Universal Gcode Sender (UGS).
      * The file streaming has completed.
      */
     void fileStreamComplete(String filename, boolean success);
-    
+
+    /**
+     * If an alarm is received from the controller
+     *
+     * @param alarm the alarm received from the controller
+     */
+    void receivedAlarm(Alarm alarm);
```

Comment:
Do you think this is needed in addition to `UGSEvent`? I sort of consider this class to be semi-deprecated, mainly because every time an event is added so many other classes break.

Reply:
Yeah, it got pretty messy to add this...

I couldn't find any other ways to notify the UGSEventListeners from the controller. Any spontaneous suggestions? Otherwise I can give it some thought.

Reply:
That's why I say semi-deprecated. Having the Controller <> Backend communication use this is fine (I'm not sure it makes sense for `UGSEvent` to originate in the Controller. But everything outside the Backend should be accessible from UGSEvent.